SimCore/include/SimCore/G4User/VolumeChecks.h

```diff
@@ -0,0 +1,115 @@
+#include "SimCore/DetectorConstruction.h"
+
+#include "Framework/Exception/Exception.h"
+#include "SimCore/SensitiveDetector.h"
+#include "SimCore/XsecBiasingOperator.h"
+
+namespace simcore {
+namespace g4user {
+namespace volumechecks {
+
+/**
+ * isInEcal
+ *
+ * Check that the passed volume is inside the ECal
+ *
+ * @TODO this is _horrible_
+ * can we get an 'ecal' and 'hcal' region instead
+ * of just a 'CalorimeterRegion' region?
+ *
+ * @param[in] vol G4LogicalVolume to check
+ * @param[in] vol_to_bias UNUSED name of volume to bias
+ */
+inline bool isInEcal(G4LogicalVolume* vol, const std::string& vol_to_bias) {
+  const G4String& volumeName = vol->GetName();
+  return ((volumeName.contains("Si") || volumeName.contains("W") ||
+           volumeName.contains("PCB") || volumeName.contains("strongback") ||
+           volumeName.contains("Glue") || volumeName.contains("CFMix") ||
+           volumeName.contains("Al") || volumeName.contains("C")) &&
+          volumeName.contains("volume")) ||
+         (volumeName.contains("nohole_motherboard"));
+}
+
+/**
+ * isInHcal
+ *
+ * Check that the passed volume is inside the HCal
+ *
+ * @param[in] vol G4LogicalVolume to check
+ * @param[in] vol_to_bias UNUSED name of volume to bias
+ */
+[[maybe_unused]] inline bool isInHcal(G4LogicalVolume* vol,
```

Comment:
I'm confused about the `[[maybe_unused]]` here and in the next occasions

Reply:
Me too, it is certainly used within `DetectorConstruction.cxx` to I think these attributes should be removed to make sure that the compilation is appropriately using this single definition in multiple places.